src/test/java/nl/altindag/log/LogCaptorShould.java

```diff
@@ -382,6 +386,45 @@ void captureMdcHeadersWhereSlf4jIsUsed() {
         assertMdcHeaders(logCaptor, "test-slf4j-mdc", "hello-slf4j");
     }
 
+    @Test
+    void detachWithTearDown() {
+        Logger logger = (Logger) LoggerFactory.getLogger(this.getClass());
+        assertThat(fetchAppenders(logger)).isEmpty();
+
+        logCaptor = LogCaptor.forClass(this.getClass());
+        assertListAppender(logger);
+
+        logCaptor.close();
+        assertThat(fetchAppenders(logger)).isEmpty();
+    }
+
+    @Test
+    void detachWithClose() {
```

Comment:
When adjusting the other function it will be a bit similar to this what, what about `detachAppenderWithAutoClosable` ?

Reply:
Done